app/assets/javascripts/mpa/datasource/GoDataSource.ts

```diff
@@ -63,52 +79,147 @@ export default class GoDataSource extends CachedDataSource<GoNameSpace, GoTerm>
-    public async getTrust(namespace: GoNameSpace = null, cutoff: number = 50, sequences: string[] = null): Promise<FATrust> {
-        if (namespace) {
-            let result: [GoTerm[], FATrust] = await this.getFromCache(namespace, Object.values(GoNameSpace), cutoff, sequences);
+    public async getTrust(namespace: GoNameSpace = null, sequences: string[] = null): Promise<FATrust> {
+        if (namespace) 
+        {
+            let result: [GoTerm[], FATrust] = await this.getFromCache(namespace, Object.values(GoNameSpace), sequences);
             return result[1];
-        } else {
-            let trusts: FATrust[] = [];
-            for (let ns of Object.values(GoNameSpace)) {
-                let result: [GoTerm[], FATrust] = await this.getFromCache(ns, Object.values(GoNameSpace), cutoff, sequences);
-                trusts.push(result[1]);
+        } 
+        else
+        {
+            if(!this._processedPeptideContainer)
+            {
+                // TODO: fix this?
+                return new FATrust(0, 0)
+            }
+
+            if(sequences == null)
+            {
+                sequences = Array.from(this._processedPeptideContainer.countTable.keys())
             }
-            return this.agregateTrust(trusts);
+
+            let totalCount = 0
+            let annotatedCount = 0
+
+            for(const seq of sequences)
+            {
+                let count = this._processedPeptideContainer.countTable.get(seq)
+                totalCount += count
+                if(this._countTable.peptide2ontology.has(seq))
+                {
+                    annotatedCount += count
+                }
+            }
+
+            return new FATrust(annotatedCount, totalCount)
         }
-        
     }
 
-    protected async computeTerms(percent = 50, sequences = null): Promise<[Map<GoNameSpace, GoTerm[]>, Map<GoNameSpace, FATrust>]> {
-        let worker = await this._repository.getWorker();
-
-        let {data, trust} = await worker.summarizeGo(percent, sequences);
-
-        let dataOutput: Map<GoNameSpace, GoTerm[]> = new Map();
-        for (let namespace of Object.values(GoNameSpace)) {
-            let items: MPAFAResult[] = data[namespace];
-            let convertedItems: GoTerm[] = [];
-            for (let item of items) {
-                let affectedPeptides: string[] = [];
-                for (let seq of Object.keys(item.sequences)) {
-                    if (item.sequences.hasOwnProperty(seq)) {
-                        for (let i = 0; i < item.sequences[seq]; i++) {
-                            affectedPeptides.push(seq);
-                        }
-                    }
+    protected async computeTerms(sequences: string[] = null): Promise<[Map<GoNameSpace, GoTerm[]>, Map<GoNameSpace, FATrust>]> 
+    {
+        // first fetch Ontology data if needed
+        var ontology: GeneOntology = this._countTable.GetOntology()
+        await ontology.fetchDefinitions(this._countTable.GetOntologyIds())
```

Comment:
Each Ontology is expected to cache fetched data, i.e. when fetchDefinitions is called with a list of ontology ids only the definitions for the ontology ids that have not yet been fetched will be fetched. E.g. this can be seen in https://github.com/unipept/unipept/blob/e864158c890f478d9f8957c3f19425f157371c7c/app/assets/javascripts/mpa/ontology/go/GeneOntology.ts#L14